browser/ts/index.ts

```diff
@@ -0,0 +1,370 @@
+    // endpoint used by Mesh. Geth & Infura both limit a request's content
+    // length to 1024 * 512 Bytes. Parity and Alchemy have much higher limits.
+    // When batch validating 0x orders, we will fit as many orders into a
+    // request without crossing the max content length. The default value is
+    // appropriate for operators using Geth or Infura. If using Alchemy or
+    // Parity, feel free to double the default max in order to reduce the number
+    // of RPC calls made by Mesh. Defaults to 524288 bytes.
+    ethereumRPCMaxContentLength?: number;
+}
+
+// The global entrypoint for creating a new MeshWrapper.
+interface ZeroExMesh {
+    newWrapperAsync(config: Config): Promise<MeshWrapper>;
+}
+
+// A direct translation of the MeshWrapper type in Go. Its API exposes only
+// simple JavaScript types like number and string, some of which will be
+// converted. For example, we will convert some strings to BigNumbers.
+interface MeshWrapper {
+    startAsync(): Promise<void>;
+    onError(handler: (err: Error) => void): void;
+    onOrderEvents(handler: (events: Array<MeshOrderEvent>) => void): void;
+    addOrdersAsync(orders: Array<MeshSignedOrder>): Promise<MeshValidationResults>;
+}
+
+// The type for signed orders exposed by MeshWrapper. Unlike other types, the
+// analog isn't defined here. Instead we re-use the definition in
+// @0x/order-utils.
+interface MeshSignedOrder {
+    makerAddress: string;
+    makerAssetData: string;
+    makerAssetAmount: string;
+    makerFee: string;
+    takerAddress: string;
+    takerAssetData: string;
+    takerAssetAmount: string;
+    takerFee: string;
+    senderAddress: string;
+    exchangeAddress: string;
+    feeRecipientAddress: string;
+    expirationTimeSeconds: string;
+    salt: string;
+    signature: string;
+}
+
+// The type for order events exposed by MeshWrapper.
+interface MeshOrderEvent {
+    orderHash: string;
+    signedOrder: MeshSignedOrder;
+    kind: string;
+    fillableTakerAssetAmount: string;
+    txHashes: Array<string>;
+}
+
+/**
+ * Order events are fired by Mesh whenever an order is added, canceled, expired,
+ * or filled.
+ */
+export interface OrderEvent {
+    orderHash: string;
+    signedOrder: SignedOrder;
+    kind: string;
+    fillableTakerAssetAmount: BigNumber;
+    txHashes: Array<string>;
+}
+
+// The type for validation results exposed by MeshWrapper.
+interface MeshValidationResults {
+    accepted: Array<MeshAcceptedOrderInfo>;
+    rejected: Array<MeshRejectedOrderInfo>;
+}
+
+// The type for accepted orders exposed by MeshWrapper.
+interface MeshAcceptedOrderInfo {
+    orderHash: string;
+    signedOrder: MeshSignedOrder;
+    fillableTakerAssetAmount: string;
+    isNew: boolean;
+}
+
+// The type for rejected orders exposed by MeshWrapper.
+interface MeshRejectedOrderInfo {
+    orderHash: string;
+    signedOrder: MeshSignedOrder;
+    kind: RejectedOrderKind;
+    status: RejectedOrderStatus;
+}
```

Comment:
These are all duplicated from `@0x/mesh-rpc-client`. Any ideas on how to avoid that? I can't think of any without creating a TS monorepo within this repo...

Reply:
I can't think of anything else either. I'm not sure if it's worth the overhead of using a TS monorepo here. Let's copy it manually for now and discuss later?